File: 812-rotate-string/rotate-string.cpp
class Solution {
public:
    void rotate(string &s){
        char ch = s[0];
        int n = s.length();
        for(int i=0;i<n-1;i++){
            s[i]=s[i+1];
        } 
        s[n-1]=ch;  
    }
    bool rotateString(string s, string goal) {
        if(s==goal) return true;
        int i=0;
        int n = s.length();
        while(i!=n){
            if(s==goal) return true;
            rotate(s);
            i++;
        }
        return false;
    }
};

File: 812-rotate-string/rotate-string.go
func rotateString(s string, goal string) bool {
    if s == goal {
        return true
    }

    n := len(s)
    if n != len(goal) {
        return false
    }

    runes := []rune(s)

    for i := 0; i < n; i++ {
        first := runes[0]
        copy(runes, runes[1:])
        runes[n-1] = first

        if string(runes) == goal {
            return true
        }
    }
    return false
}
